*** fs/romfs/Config.in.orig	2011-11-29 11:05:36.000000000 -0800
--- fs/romfs/Config.in	2011-11-29 11:41:51.000000000 -0800
***************
*** 3,5 ****
--- 3,40 ----
  	help
  	  Build a romfs image of the root filesystem.
  
+ choice
+ 	prompt "Compression method"
+ 	default BR2_TARGET_ROOTFS_ROMFS_NONE
+ 	depends on BR2_TARGET_ROOTFS_ROMFS
+ 	help
+ 	  Select compressor for ext2 filesystem of the root filesystem
+ 
+ config BR2_TARGET_ROOTFS_ROMFS_NONE
+ 	bool "no compression"
+ 	help
+ 	  Do not compress the ext2 filesystem.
+ 
+ config BR2_TARGET_ROOTFS_ROMFS_GZIP
+ 	bool "gzip"
+ 	help
+ 	  Do compress the ext2 filesystem with gzip.
+ 
+ config BR2_TARGET_ROOTFS_ROMFS_BZIP2
+ 	bool "bzip2"
+ 	help
+ 	  Do compress the ext2 filesystem with bzip2.
+ 
+ config BR2_TARGET_ROOTFS_ROMFS_LZMA
+ 	bool "lzma"
+ 	help
+ 	  Do compress the ext2 filesystem with lzma.
+ 
+ endchoice
+ 
+ config BR2_TARGET_ROOTFS_ROMFS_MKIMAGE
+ 	bool "make uboot image of romfs"
+ 	depends on BR2_TARGET_ROOTFS_ROMFS
+ 	help
+ 	  Create a image file with header information for uboot (using mkimage).
+ 
*** fs/ext2/Config.in.orig	2011-11-29 11:54:16.000000000 -0800
--- fs/ext2/Config.in	2011-11-29 11:55:09.000000000 -0800
***************
*** 47,49 ****
--- 47,55 ----
  
  endchoice
  
+ config BR2_TARGET_ROOTFS_EXT2_MKIMAGE
+ 	bool "make uboot image of ext2 filesystem"
+ 	depends on BR2_TARGET_ROOTFS_EXT2
+ 	help
+ 	  Create a image file with header information for uboot (using mkimage).
+ 
*** fs/common.mk.orig	2011-11-29 11:11:15.000000000 -0800
--- fs/common.mk	2011-11-29 11:58:39.000000000 -0800
***************
*** 39,45 ****
  define ROOTFS_TARGET_INTERNAL
  
  # extra deps
! $(eval ROOTFS_$(2)_DEPENDENCIES += host-fakeroot host-makedevs $(if $(BR2_TARGET_ROOTFS_$(2)_LZMA),host-lzma))
  
  $(BINARIES_DIR)/rootfs.$(1): $(ROOTFS_$(2)_DEPENDENCIES)
  	@$(call MESSAGE,"Generating root filesystem image rootfs.$(1)")
--- 39,45 ----
  define ROOTFS_TARGET_INTERNAL
  
  # extra deps
! $(eval ROOTFS_$(2)_DEPENDENCIES += host-fakeroot host-makedevs $(if $(BR2_TARGET_ROOTFS_$(2)_LZMA),host-lzma) $(if $(BR2_TARGET_ROOTFS_$(2)_MKIMAGE),host-uboot-tools))
  
  $(BINARIES_DIR)/rootfs.$(1): $(ROOTFS_$(2)_DEPENDENCIES)
  	@$(call MESSAGE,"Generating root filesystem image rootfs.$(1)")
***************
*** 59,70 ****
--- 59,84 ----
  	$(foreach hook,$(ROOTFS_$(2)_POST_GEN_HOOKS),$(call $(hook))$(sep))
  ifeq ($$(BR2_TARGET_ROOTFS_$(2)_GZIP),y)
  	gzip -9 -c $$@ > $$@.gz
+ ifeq ($$(BR2_TARGET_ROOTFS_$(2)_MKIMAGE),y)
+ 	$(HOST_DIR)/usr/bin/mkimage -A $(patsubst i%86,x86,$(ARCH)) -O linux -T ramdisk -C gzip -a 0 -e 0 -d $$@.gz $$@.gz.uimg
+ endif
  endif
  ifeq ($$(BR2_TARGET_ROOTFS_$(2)_BZIP2),y)
  	bzip2 -9 -c $$@ > $$@.bz2
+ ifeq ($$(BR2_TARGET_ROOTFS_$(2)_MKIMAGE),y)
+ 	$(HOST_DIR)/usr/bin/mkimage -A $(patsubst i%86,x86,$(ARCH)) -O linux -T ramdisk -C bzip2 -a 0 -e 0 -d $$@.bz2 $$@.bz2.uimg
+ endif
  endif
  ifeq ($$(BR2_TARGET_ROOTFS_$(2)_LZMA),y)
  	$(LZMA) -9 -c $$@ > $$@.lzma
+ ifeq ($$(BR2_TARGET_ROOTFS_$(2)_MKIMAGE),y)
+ 	$(HOST_DIR)/usr/bin/mkimage -A $(patsubst i%86,x86,$(ARCH)) -O linux -T ramdisk -C lzma -a 0 -e 0 -d $$@.lzma $$@.lzma.uimg
+ endif
+ endif
+ ifeq ($$(BR2_TARGET_ROOTFS_$(2)_NONE),y)
+ ifeq ($$(BR2_TARGET_ROOTFS_$(2)_MKIMAGE),y)
+ 	$(HOST_DIR)/usr/bin/mkimage -A $(patsubst i%86,x86,$(ARCH)) -O linux -T ramdisk -C none -a 0 -e 0 -d $$@ $$@.uimg
+ endif
  endif
  
  rootfs-$(1)-show-depends:
