*** toolchain/gdb/gdb.mk.orig	2013-04-30 06:47:43.487204000 -0700
--- toolchain/gdb/gdb.mk	2013-04-30 07:45:50.511089000 -0700
***************
*** 182,187 ****
--- 182,188 ----
  		$(GDB_DIR)/configure $(QUIET) \
  		--cache-file=/dev/null \
  		--prefix=$(STAGING_DIR) \
+ 		--with-sysroot=$(STAGING_DIR)/../../../../target \
  		--build=$(GNU_HOST_NAME) \
  		--host=$(GNU_HOST_NAME) \
  		--target=$(REAL_GNU_TARGET_NAME) \
