*** linux-4.19/arch/arm/boot/dts/zynq-fpga.dtsi.orig	2018-11-06 14:58:38.574225179 -0800
--- linux-4.19/arch/arm/boot/dts/zynq-fpga.dtsi	2018-11-06 15:02:45.640883517 -0800
***************
*** 0 ****
--- 1,115 ----
+ /*
+  *  Copyright (C) 2011 - 2014 Xilinx
+  *  Copyright (C) 2012 National Instruments Corp.
+  *
+  * This software is licensed under the terms of the GNU General Public
+  * License version 2, as published by the Free Software Foundation, and
+  * may be copied, distributed, and modified under those terms.
+  *
+  * This program is distributed in the hope that it will be useful,
+  * but WITHOUT ANY WARRANTY; without even the implied warranty of
+  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  * GNU General Public License for more details.
+  */
+ 
+ / {
+ 	amba {
+ 
+ 		ps7-axislv@40000000 {
+ 			compatible = "generic-uio";
+ 			reg = <0x40000000 0x40000000>;
+ 			interrupt-parent = <&intc>;
+ 		};
+ 
+ 		ps7-axislv0@43c00000 {
+ 			compatible = "generic-uio";
+ 			reg = <0x43c00000 0x1000>;
+ 			interrupt-parent = <&intc>;
+ 
+ 			/* must subtract 32 from zynq irq ID# PL interrupts are
+ 			 * in two slices: 61...68 and 84..91
+ 			 */
+ 			interrupts = <0 29 4>;
+ 		};
+ 
+ 		ps7-axislv1@43c01000 {
+ 			compatible = "generic-uio";
+ 			reg = <0x43c01000 0x1000>;
+ 			interrupt-parent = <&intc>;
+ 			interrupts = <0 30 4>;
+ 		};
+ 
+ 		ps7-axislv2@43c02000 {
+ 			compatible = "generic-uio";
+ 			reg = <0x43c02000 0x1000>;
+ 			interrupt-parent = <&intc>;
+ 			interrupts = <0 31 4>;
+ 		};
+ 
+ 		ps7-axislv3@43c03000 {
+ 			compatible = "generic-uio";
+ 			reg = <0x43c03000 0x1000>;
+ 			interrupt-parent = <&intc>;
+ 			interrupts = <0 32 4>;
+ 		};
+ 
+ 		ps7-axislv4@43c04000 {
+ 			compatible = "generic-uio";
+ 			reg = <0x43c04000 0x1000>;
+ 			interrupt-parent = <&intc>;
+ 			interrupts = <0 33 4>;
+ 		};
+ 
+ 		ps7-axislv5@43c05000 {
+ 			compatible = "generic-uio";
+ 			reg = <0x43c05000 0x1000>;
+ 			interrupt-parent = <&intc>;
+ 			interrupts = <0 34 4>;
+ 		};
+ 
+ 		ps7-axislv6@43c06000 {
+ 			compatible = "generic-uio";
+ 			reg = <0x43c06000 0x1000>;
+ 			interrupt-parent = <&intc>;
+ 			interrupts = <0 35 4>;
+ 		};
+ 
+ 		ps7-axislv7@43c07000 {
+ 			compatible = "generic-uio";
+ 			reg = <0x43c07000 0x1000>;
+ 			interrupt-parent = <&intc>;
+ 			interrupts = <0 36 4>;
+ 		};
+ 
+ 		ps7-axislv8@43c08000 {
+ 			compatible = "generic-uio";
+ 			reg = <0x43c08000 0x1000>;
+ 			interrupt-parent = <&intc>;
+ 			interrupts = <0 52 4 >;
+ 		};
+ 
+ 		ps7-axislv9@43c09000 {
+ 			compatible = "generic-uio";
+ 			reg = <0x43c09000 0x1000>;
+ 			interrupt-parent = <&intc>;
+ 			interrupts = <0 53 4 >;
+ 		};
+ 
+ 		ps7-axislv10@43c20000 {
+ 			compatible = "generic-uio";
+ 			reg = <0x43c20000 0x20000>;
+ 			interrupt-parent = <&intc>;
+ 			interrupts = <0 54 4 >;
+ 		};
+ 	};
+ 
+ 	prog_fpga0: prog-fpga0 {
+ 		compatible = "tills,fpga-programmer-1.0";
+ 		fpga-mgr   = <&devcfg>;
+ 		file       = "zynq-firmware.bin";
+ 	};
+ };
+ 
+ &clkc {
+ 	fclk-enable      = <0xf>;
+ };
*** linux-4.19/arch/arm/boot/dts/zynq-zc706.dts.orig	2018-11-06 15:08:30.441190461 -0800
--- linux-4.19/arch/arm/boot/dts/zynq-zc706.dts	2018-11-06 15:08:48.784597455 -0800
***************
*** 5,10 ****
--- 5,11 ----
   */
  /dts-v1/;
  #include "zynq-7000.dtsi"
+ #include "zynq-fpga.dtsi"
  
  / {
  	model = "Xilinx ZC706 board";
***************
*** 322,324 ****
--- 323,329 ----
  	pinctrl-names = "default";
  	pinctrl-0 = <&pinctrl_usb0_default>;
  };
+ 
+ &i2c0 {
+ 	status = "okay";
+ };
*** linux-4.19/arch/arm/boot/dts/zynq-zybo.dts.orig	2018-10-21 23:37:37.000000000 -0700
--- linux-4.19/arch/arm/boot/dts/zynq-zybo.dts	2018-11-06 15:05:05.915982290 -0800
***************
*** 5,10 ****
--- 5,11 ----
   */
  /dts-v1/;
  #include "zynq-7000.dtsi"
+ #include "zynq-fpga.dtsi"
  
  / {
  	model = "Digilent Zybo board";
***************
*** 61,63 ****
--- 62,68 ----
  	dr_mode = "host";
  	usb-phy = <&usb_phy0>;
  };
+ 
+ &i2c0 {
+ 	status = "okay";
+ };
