--- busybox-1.31.0/init/init.c.orig	2024-11-08 18:01:42.525802953 -0800
+++ busybox-1.31.0/init/init.c	2024-11-08 18:02:27.896813131 -0800
@@ -1074,6 +1074,27 @@
 		sigaction(SIGBUS, &sa, NULL);
 	}
 #endif
+	
+    /* Set some hard limits to 'infinity' (for everyone to inherit).
+	 * We want real-time related stuff to be used by everyone (but
+	 * not enabled unless soft-limit is increased from the shell).
+	 * We also set coredump hard-limit here so coredumps may be
+	 * enabled by everyone.
+	 * T.S; 2012/3/30.
+	 */
+	{
+		struct rlimit limit;
+
+		/* Keep current soft limit for RLIMIT_MEMLOCK */
+		if ( 0 == getrlimit( RLIMIT_MEMLOCK, &limit ) ) {
+			limit.rlim_max = RLIM_INFINITY;
+			setrlimit( RLIMIT_MEMLOCK, &limit );
+		}
+		limit.rlim_cur = 0;
+		limit.rlim_max = RLIM_INFINITY;
+		setrlimit( RLIMIT_RTPRIO, &limit );
+		setrlimit( RLIMIT_CORE,   &limit );
+	}
 
 	if (!DEBUG_INIT) {
 		/* Some users send poweroff signals to init VERY early.
