*** package/e2fsprogs/e2fsprogs.mk.orig	2017-09-05 23:33:31.900828741 -0700
--- package/e2fsprogs/e2fsprogs.mk	2017-09-05 23:35:43.697192869 -0700
***************
*** 32,37 ****
--- 32,38 ----
  	--disable-fuse2fs \
  	--disable-libblkid \
  	--disable-libuuid \
+ 	--enable-symlink-build \
  	--enable-symlink-install \
  	--disable-testio-debug
  
***************
*** 52,57 ****
--- 53,59 ----
  	--disable-e2initrd-helper \
  	--disable-testio-debug \
  	--disable-rpath \
+ 	--enable-symlink-build \
  	--enable-symlink-install
  
  ifeq ($(BR2_PACKAGE_E2FSPROGS_FUSE2FS),y)
