*** package/e2fsprogs/e2fsprogs.mk.orig	2014-09-15 09:50:43.000000000 -0700
--- package/e2fsprogs/e2fsprogs.mk	2014-09-15 09:51:25.000000000 -0700
***************
*** 22,30 ****
--- 22,36 ----
  	--disable-libuuid \
  	--enable-fsck \
  	--disable-e2initrd-helper \
+  	--enable-symlink-build \
+  	--enable-symlink-install \
  	--disable-testio-debug \
  	--disable-rpath
  
+ HOST_E2FSPROGS_CONF_OPT = \
+  	--enable-symlink-build \
+  	--enable-symlink-install
+  
  ifeq ($(BR2_nios2),y)
  E2FSPROGS_CONF_ENV += ac_cv_func_fallocate=no
  endif
