--- package/ncurses/ncurses.mk.orig	2025-04-11 13:35:22.221802888 -0700
+++ package/ncurses/ncurses.mk	2025-04-11 13:37:20.205520510 -0700
@@ -28,6 +28,7 @@
 	--disable-rpath \
 	--disable-rpath-hack \
 	--enable-echo \
+	--enable-symlinks \
 	--enable-const \
 	--enable-overwrite \
 	--enable-pc-files \
@@ -157,6 +158,7 @@
 	--without-gpm \
 	--without-manpages \
 	--without-cxx \
+	--enable-symlinks \
 	--without-cxx-binding \
 	--without-ada \
 	--with-default-terminfo-dir=/usr/share/terminfo \
