*** support/scripts/check-kernel-headers.sh.orig	2014-10-09 19:40:02.000000002 -0700
--- support/scripts/check-kernel-headers.sh	2014-10-09 19:40:16.000000002 -0700
***************
*** 8,14 ****
  HDR_V="${HDR_VER#*.}"
  HDR_m="${HDR_V%%.*}"
  
! EXEC="$(mktemp --tmpdir check-headers.XXXXXX)"
  
  # We do not want to account for the patch-level, since headers are
  # not supposed to change for different patchlevels, so we mask it out.
--- 8,14 ----
  HDR_V="${HDR_VER#*.}"
  HDR_m="${HDR_V%%.*}"
  
! EXEC="$(mktemp -t check-headers.XXXXXX)"
  
  # We do not want to account for the patch-level, since headers are
  # not supposed to change for different patchlevels, so we mask it out.
