--- package/kmod/kmod.mk.orig	2025-04-11 13:16:31.798750642 -0700
+++ package/kmod/kmod.mk	2025-04-11 13:20:49.884702011 -0700
@@ -4,12 +4,23 @@
 #
 ################################################################################
 
+# T.S: testsuite contains hard links which are not supported under AFS
+#  argument 1 is the uppercase package name, including an HOST_ prefix
+#             for host packages
+define kmod-extract
+      $(INFLATE$(suffix $(KMOD_SOURCE))) $(KMOD_DL_DIR)/$(KMOD_SOURCE) | \
+      $(TAR) --strip-components=1 --exclude=testsuite -C $($(1)_DIR) $(TAR_OPTIONS) -
+endef
+
+HOST_KMOD_EXTRACT_CMDS = $(call kmod-extract,HOST_KMOD)
+KMOD_EXTRACT_CMDS = $(call kmod-extract,KMOD)
+
 KMOD_VERSION = 33
 KMOD_SOURCE = kmod-$(KMOD_VERSION).tar.xz
 KMOD_SITE = $(BR2_KERNEL_MIRROR)/linux/utils/kernel/kmod
 KMOD_INSTALL_STAGING = YES
 KMOD_DEPENDENCIES = host-pkgconf
-HOST_KMOD_DEPENDENCIES = host-pkgconf
+HOST_KMOD_DEPENDENCIES = host-pkgconf host-zlib
 
 # license info for libkmod only, conditionally add more below
 KMOD_LICENSE = LGPL-2.1+ (library)
@@ -28,7 +39,7 @@
 KMOD_CONF_OPTS = --disable-static --enable-shared
 
 KMOD_CONF_OPTS += --disable-manpages
-HOST_KMOD_CONF_OPTS = --disable-manpages
+HOST_KMOD_CONF_OPTS = --disable-manpages --with-zlib
 
 ifeq ($(BR2_PACKAGE_BASH_COMPLETION),y)
 KMOD_CONF_OPTS += --with-bashcompletiondir=/usr/share/bash-completion/completions
