*** package/glibc/glibc.mk.orig	2021-08-12 23:02:10.368430790 +0200
--- package/glibc/glibc.mk	2021-08-12 23:05:42.246869721 +0200
***************
*** 146,152 ****
  define GLIBC_INSTALL_TARGET_CMDS
  	for libpattern in $(GLIBC_LIBS_LIB); do \
  		$(call copy_toolchain_lib_root,$$libpattern) ; \
! 	done
  endef
  
  $(eval $(autotools-package))
--- 146,153 ----
  define GLIBC_INSTALL_TARGET_CMDS
  	for libpattern in $(GLIBC_LIBS_LIB); do \
  		$(call copy_toolchain_lib_root,$$libpattern) ; \
! 	done ; \
! 	$(INSTALL) -D -m0755 $(STAGING_DIR)/sbin/ldconfig $(TARGET_DIR)/sbin/
  endef
  
  $(eval $(autotools-package))
