*** package/gcc/gcc-final/gcc-final.mk.orig	2015-03-01 13:26:12.000000000 -0800
--- package/gcc/gcc-final/gcc-final.mk	2015-04-30 13:53:28.990090867 -0700
***************
*** 44,49 ****
--- 44,50 ----
  # Languages supported by the cross-compiler
  GCC_FINAL_CROSS_LANGUAGES-y = c
  GCC_FINAL_CROSS_LANGUAGES-$(BR2_INSTALL_LIBSTDCPP) += c++
+ GCC_FINAL_CROSS_LANGUAGES-$(BR2_INSTALL_FORTRAN)   += fortran
  GCC_FINAL_CROSS_LANGUAGES = $(subst $(space),$(comma),$(GCC_FINAL_CROSS_LANGUAGES-y))
  
  HOST_GCC_FINAL_CONF_OPTS = \
***************
*** 130,135 ****
--- 131,141 ----
  HOST_GCC_FINAL_USR_LIBS += libgomp
  endif
  
+ ifeq ($(BR2_INSTALL_FORTRAN),y)
+ HOST_GCC_FINAL_USR_LIBS += libgfortran
+ HOST_GCC_FINAL_USR_LIBS += libquadmath
+ endif
+ 
  ifeq ($(BR2_GCC_ENABLE_LIBMUDFLAP),y)
  ifeq ($(BR2_TOOLCHAIN_HAS_THREADS),y)
  HOST_GCC_FINAL_USR_LIBS += libmudflapth
***************
*** 141,158 ****
  ifneq ($(HOST_GCC_FINAL_USR_LIBS),)
  define HOST_GCC_FINAL_INSTALL_STATIC_LIBS
  	for i in $(HOST_GCC_FINAL_USR_LIBS) ; do \
! 		cp -dpf $(HOST_DIR)/usr/$(GNU_TARGET_NAME)/lib*/$${i}.a \
! 			$(STAGING_DIR)/usr/lib/ ; \
  	done
  endef
  
  ifeq ($(BR2_STATIC_LIBS),)
  define HOST_GCC_FINAL_INSTALL_SHARED_LIBS
  	for i in $(HOST_GCC_FINAL_USR_LIBS) ; do \
! 		cp -dpf $(HOST_DIR)/usr/$(GNU_TARGET_NAME)/lib*/$${i}.so* \
! 			$(STAGING_DIR)/usr/lib/ ; \
! 		cp -dpf $(HOST_DIR)/usr/$(GNU_TARGET_NAME)/lib*/$${i}.so* \
! 			$(TARGET_DIR)/usr/lib/ ; \
  	done
  endef
  endif
--- 147,170 ----
  ifneq ($(HOST_GCC_FINAL_USR_LIBS),)
  define HOST_GCC_FINAL_INSTALL_STATIC_LIBS
  	for i in $(HOST_GCC_FINAL_USR_LIBS) ; do \
! 		if ! cp -dpf $(HOST_DIR)/usr/$(GNU_TARGET_NAME)/lib*/$${i}.a \
! 				$(STAGING_DIR)/usr/lib/ 2>/dev/null ; then \
! 			echo "WARNING: $(HOST_DIR)/usr/$(GNU_TARGET_NAME)/lib*/$${i}.a not found" ; \
! 		fi \
  	done
  endef
  
  ifeq ($(BR2_STATIC_LIBS),)
  define HOST_GCC_FINAL_INSTALL_SHARED_LIBS
  	for i in $(HOST_GCC_FINAL_USR_LIBS) ; do \
! 		if ! cp -dpf $(HOST_DIR)/usr/$(GNU_TARGET_NAME)/lib*/$${i}.so* \
! 				$(STAGING_DIR)/usr/lib/ 2>/dev/null ; then \
! 			echo "WARNING: $(HOST_DIR)/usr/$(GNU_TARGET_NAME)/lib*/$${i}.a not found" ; \
! 		fi ; \
! 		if ! cp -dpf $(HOST_DIR)/usr/$(GNU_TARGET_NAME)/lib*/$${i}.so* \
! 				$(TARGET_DIR)/usr/lib/ 2>/dev/null ; then \
! 			echo "WARNING: $(HOST_DIR)/usr/$(GNU_TARGET_NAME)/lib*/$${i}.a not found" ; \
! 		fi \
  	done
  endef
  endif
