*** dropbear-2012.55/options.h.orig	2012-06-18 09:43:15.000000000 -0500
--- dropbear-2012.55/options.h	2012-06-18 09:43:27.000000000 -0500
***************
*** 180,186 ****
   * Public key logins are allowed for blank-password accounts regardless of this
   * setting.  PAM is not affected by this setting, it uses the normal pam.d
   * settings ('nullok' option) */
! /* #define ALLOW_BLANK_PASSWORD */
  
  #define ENABLE_CLI_PASSWORD_AUTH
  #define ENABLE_CLI_PUBKEY_AUTH
--- 180,186 ----
   * Public key logins are allowed for blank-password accounts regardless of this
   * setting.  PAM is not affected by this setting, it uses the normal pam.d
   * settings ('nullok' option) */
! #define ALLOW_BLANK_PASSWORD
  
  #define ENABLE_CLI_PASSWORD_AUTH
  #define ENABLE_CLI_PUBKEY_AUTH
