*** dropbear-2012.55/options.h.orig	2012-07-11 11:50:36.000000000 -0500
--- dropbear-2012.55/options.h	2012-07-11 11:50:50.000000000 -0500
***************
*** 52,58 ****
  /*#define DROPBEAR_SMALL_CODE*/
  
  /* Enable X11 Forwarding - server only */
! #define ENABLE_X11FWD
  
  /* Enable TCP Fowarding */
  /* 'Local' is "-L" style (client listening port forwarded via server)
--- 52,58 ----
  /*#define DROPBEAR_SMALL_CODE*/
  
  /* Enable X11 Forwarding - server only */
! /* #define ENABLE_X11FWD */
  
  /* Enable TCP Fowarding */
  /* 'Local' is "-L" style (client listening port forwarded via server)
***************
*** 179,185 ****
  /* Define this to allow logging in to accounts that have no password specified.
   * Public key logins are allowed for blank-password accounts regardless of this
   * setting. */
! /* #define ALLOW_BLANK_PASSWORD */
  
  #define ENABLE_CLI_PASSWORD_AUTH
  #define ENABLE_CLI_PUBKEY_AUTH
--- 179,185 ----
  /* Define this to allow logging in to accounts that have no password specified.
   * Public key logins are allowed for blank-password accounts regardless of this
   * setting. */
! #define ALLOW_BLANK_PASSWORD
  
  #define ENABLE_CLI_PASSWORD_AUTH
  #define ENABLE_CLI_PUBKEY_AUTH
