*** package/dropbear/S50dropbear.orig	2013-03-14 10:48:45.528709000 -0700
--- package/dropbear/S50dropbear	2013-03-14 10:49:02.016977000 -0700
***************
*** 24,30 ****
  		/usr/bin/dropbearkey -t dss -f /etc/dropbear/dropbear_dss_host_key > /dev/null 2>&1
  	fi
  	umask 077
! 	start-stop-daemon -S -q -p /var/run/dropbear.pid --exec /usr/sbin/dropbear
  	echo "OK"
  }
  stop() {
--- 24,30 ----
  		/usr/bin/dropbearkey -t dss -f /etc/dropbear/dropbear_dss_host_key > /dev/null 2>&1
  	fi
  	umask 077
! 	start-stop-daemon -S -q -p /var/run/dropbear.pid --exec /usr/sbin/dropbear -- -w
  	echo "OK"
  }
  stop() {
