*** ./package/kmod/kmod.mk.orig	2014-04-01 08:09:54.000000002 -0700
--- ./package/kmod/kmod.mk	2014-04-01 13:27:08.000000002 -0700
***************
*** 15,20 ****
--- 15,31 ----
  KMOD_LICENSE = LGPLv2.1+
  KMOD_LICENSE_FILES = libkmod/COPYING
  
+ # T.S: testsuite contains hard links which are not supported under AFS
+ #  argument 1 is the uppercase package name, including an HOST_ prefix
+ #             for host packages
+ define kmod-extract
+ 	$(INFLATE$(suffix $(KMOD_SOURCE))) $(DL_DIR)/$(KMOD_SOURCE) | \
+ 	$(TAR) $(TAR_STRIP_COMPONENTS)=1 --exclude=testsuite -C $($(1)_DIR) $(TAR_OPTIONS) -
+ endef
+ 
+ HOST_KMOD_EXTRACT_CMDS = $(call kmod-extract,HOST_KMOD)
+ KMOD_EXTRACT_CMDS = $(call kmod-extract,KMOD)
+ 
  # static linking not supported, see
  # https://git.kernel.org/cgit/utils/kernel/kmod/kmod.git/commit/?id=b7016153ec8
  KMOD_CONF_OPT = --disable-static --enable-shared
