*** dropbear-983a817f8e41/options.h.orig	2012-06-11 08:55:14.214437000 -0700
--- dropbear-983a817f8e41/options.h	2012-06-11 08:55:20.133533000 -0700
***************
*** 197,203 ****
  /* Define this to allow logging in to accounts that have no password specified.
   * Public key logins are allowed for blank-password accounts regardless of this
   * setting. */
! /* #define ALLOW_BLANK_PASSWORD */
  
  #define ENABLE_CLI_PASSWORD_AUTH
  #define ENABLE_CLI_PUBKEY_AUTH
--- 197,203 ----
  /* Define this to allow logging in to accounts that have no password specified.
   * Public key logins are allowed for blank-password accounts regardless of this
   * setting. */
! #define ALLOW_BLANK_PASSWORD
  
  #define ENABLE_CLI_PASSWORD_AUTH
  #define ENABLE_CLI_PUBKEY_AUTH
